components/playlist/browser/resources/media_detector/youtube.com.js

```diff
@@ -11,15 +11,23 @@ const siteSpecificDetector = { // eslint-disable-line no-unused-vars
       // The last one is known to be one with the hight resolution
       thumbnail = thumbnail[thumbnail.length - 1]?.url
     }
-    return thumbnail
+    return (typeof(thumbnail) === 'string' && (thumbnail.startsWith('/') || thumbnail.startsWith('http://') || thumbnail.startsWith('https://'))) 
+        ? thumbnail : null
   },
   getMediaTitle: function (node) {
-    return window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.title
+    const title = window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.title
+    return typeof(title) === 'string' ? title : null
   },
   getMediaAuthor: function (node) {
-    return window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.author
+    const author = window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.author
+    return typeof(author) === 'string' ? author : null
   },
   getMediaDurationInSeconds: function (node) {
-    return window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.lengthSeconds
+    let duration = window.ytplayer?.bootstrapPlayerResponse?.videoDetails?.lengthSeconds
+    if (typeof(duration) == 'string') {
+      duration = Number.parseFloat(duration)
+      return Number.isNaN(duration) ? null : duration
```

Comment:
@sangwoo108 just curious what would happen on the native side if this checks passes but `duration` is actually not a number? A page can redefine `Number.isNaN()` to always return `true`

Reply:
Oh, good question. I think it's okay as we are filtering types from C++ side once more. If the duration isn't number, `    auto duration = media_dict.FindInt("duration");` will be empty value. So the duration will remain 0.